src/components/flat-table/flat-table-row/flat-table-row.component.tsx

```diff
@@ -105,8 +105,14 @@ export const FlatTableRow = React.forwardRef<
     const [firstCellId, setFirstCellId] = useState<string | null>(null);
     const [cellsArray, setCellsArray] = useState<Element[]>([]);
     const [tabIndex, setTabIndex] = useState(-1);
+    const [rowHeight, setRowHeight] = useState(0);
     let interactiveRowProps = {};
 
+    useLayoutEffect(() => {
```

Comment:
**comment** (non-blocking): I'm not a big fan of using state to update the height, and especially not when that is only updated in an effect that runs only once when the component is first rendered.

I don't think there's a perfect solution here, because in order to catch any changes in height (eg due to the user resizing the window, when the cell has enough text to run to several lines), we'd ideally need to use a `ResizeObserver`, which is rather more complex than I'd like to do. (And this behaviour never worked correctly anyway, it's certainly not a priority.)

My suggestion of using the `ref` directly for the `rowHeight` prop (see suggestion below) has a slight advantage over your solution in at least (potentially) updating the height on every render, so if this is in a table that dynamically changes quite a bit then there's more chance of the focus styles being adjusted to the correct height where things change. On the other hand, there might be performance implications (I doubt it, reading the `offsetHeight` shouldn't be expensive, but I can't guarantee that it's never an issue), so I'm not suggesting which way is better.

Reply:
The performance side of things is a valid consideration. I'd like to think that if Flat Table is used and it has to render that much data, it would be done so using a Pager component to limit the amount of rows rendered to 10, 25 etc.